text/1006-groupped-cli.rst

```diff
@@ -0,0 +1,240 @@
+the "tool within a tool" concept without reading RFCs or the documentation.
+Most likely the CLI would be just perceived as inconsistent at best, or
+inconvinient in the worst case.
+
+With adding the ``edgedb project`` subcommand, it became clear that maintaining
+two entirely different naming schemes in one tool will be confusing to the
+users.  Here's what a portion of ``--help`` output would look like if we keep
+the current approach::
+
+    <...>
+
+    Migrations:
+        create-migration         Create a migration script
+        apply-migrations         Bring current database to the latest or
+                                 a specified revision
+        show-migrations          Show all migration versions
+        check-migrations         Show current migration state
+
+    Server Management:
+        server status            Status of an instance
+        server init              Initialize a new server instance
+        server start             Start an instance
+        server stop              Stop an instance
+        server restart           Restart an instance
+        server info              Show server information
+
+    <...>
+
+The actual ``--help`` output will be considerably longer than that. It will
+likely take a few days for the users to not be confused whether to type
+``edgedb project init`` or ``edgedb init-project``, because they have just
+run ``edgedb create-migration``.
+
+Lastly, we will inevitably continue evolving the EdgeDB CLI by adding new
+subcommands. For example, we might add the ``edgedb cloud`` command group which
+would classify as a "tool within a tool". Or we might add subcommands to manage
+authentication and other aspects of database management, which are unlikely
+to be classified as "tools within a tool", in which case we will have more
+commands in the form of ``edgedb create-X`` or ``edgedb show-Y``. Ultimately,
+the number of commands with entirely different naming will only continue to
+grow.
+
+
+Overview
+========
+
+The RFC proposes to have the following comamnds structure for the
+the EdgeDB RC1 release::
+
+    dump                       Create a database backup
+    restore                    Restore a database backup from file
+    configure                  Configure a DB or a server instance
+
+    migration apply            Migrate the database to the latest revision
+    migration create           Create a new migration
+    migration log              Show the migrations log
+    migration status           Show the current migration state
+    migrate                    An alias for `edgedb migration apply`
+
+    project init               Initialize a new EdgeDB project
+    project status             Show the status of the current project
+    project list               List all projects
+    project unlink             Clean-up the project configuration
+
+    server status              Status of an instance
```

Comment:
### Thoughts

`edgedb server status` without instance name feels more like `edgedb server list-instances` intuitively, especially when we also have `edgedb server info` (and `edgedb server list-versions`) which sounds similar but actually shows info about the installations. Even though I know it's an over-optimization to suggest a new `edgedb instance` group for all instance-related subcommands (because instances depend on installations, and they are all "servers" literally), but `edgedb server status` keeps pushing me towards that direction.

### Proposal

Perhaps `edgedb server list-instances` could be an alias of {`edgedb server status` without instance name}?

Reply:
The initial plan was to have `server status --all`. But it turned out that `status` without name is useless and could print a list instead of erroring out. Also we expect most of the users having a small number of instances (probably single one). So it doesn't make sense to learn two commands.

We may have `list-instances` that is faster as does no status check for scripting.